lib/Subscriber/Core/RegistratorSubscriber.php

```diff
@@ -98,6 +108,15 @@ public function handleDocumentFromRegistry(HydrateEvent $event)
         $document = $this->documentRegistry->getDocumentForNode($node);
 
         $event->setDocument($document);
+
+        $options = $event->getOptions();
+
+        // if reydration is not required (f.e. we just want to retrieve the
+        // current state of the document, no matter it's current state) stop
+        // further event propagation - we have the document now.
+        if (isset($options['rehydrate']) && false === $options['rehydrate']) {
```

Comment:
Note that the forced use of `isset` is a bug. Caused by the fact that options are not inherited by proxies, making it possible for an empty array to be passed as `$options`. This is partially addressed by @wachterjohannes 's fix, but not completely apparently. I think there is a better way, see #45